### ISSUE TITLE
bulk_create with empty new_models

### ISSUE THREAD

**Issue by mczerski** [mathandpencil/django-bulk-sync#11]
hi,

it seems that it is not possible to use bulk_create to handle case when there should be only deletes.
when new_models is empty I get this exception:
  File "../bulk_sync/__init__.py", line 22, in bulk_sync
    db_class = new_models[0].__class__
IndexError: list index out of range

**Comment by ses4j** [comment]
Ah, interesting.  If the invocation is:

    ret = bulk_sync(new_models=[], filters=None, key_fields=("name",), skip_updates=True)

Then it doesn't know what Django model to query/delete.  As a way forward, I can see:

- make bulk_sync a Manager thing, so it'd know the object by the class it was invoked on.
- If you give it `new_models=MyModel.objects.none()`, that could be made to work though would not as coded.
- allow a required or optional param to pass the Django model you want to operate on.

Thoughts?

**Comment by mczerski** [comment]
Optional param was what i first thought about. I think this is the most obvious API from the user perspective.

**Comment by morugu** [comment]
same issue.

@ses4j @mczerski 
I think optional param is best.
If It's possible that new_models is empty, user set model param.

ex.
```
bulk_sync(new_models=[], key_fields=['name'], filters=None, model=MyModel())
```

**Comment by ses4j** [comment]
@morugu @mczerski Committed to master.  If that looks good I'll release.

**Comment by ses4j** [comment]
Released in v3.1.0.
